### DISCUSSION TITLE
How to draw waveform on cliend side on demand?

### DISCUSSION THREAD

**Discussion by ApayRus** [katspaugh/wavesurfer.js#2950] [Q&A] [answered]
Hi. I use wavesurfer at first as a phrasal player, to play regions in a book (connected to phrases/regions in audio). 
I do `wavesurfer.load(mediaElement, [])`. So, wafesurfer is `ready` fast and  user can play phrases. 
If I do `wavesurfer.load(mediaElement)`, wavesurfer will be calculating peaks, that user don't need yet... 
But sometimes user want to open waveform, to change regions (subtitles). In that case I need to calculate peaks and render them. 
I know how to get peaks from other libraries, and then setPeaks on wavesurfer. But how to calculate peaks with wavesurfer on demand, not by default?

**Comment by katspaugh** [comment, marked as answer]
As you suggested yourself, you can first load the audio with dummy peaks, and then render it by calling `load` again w/o peaks.

See this example: https://wavesurfer-js.org/examples/#9a2f6356aa8fcf3a87ec9cf3bb6cda6c

Please note that in this case the audio will potentially load twice, whereas if you just call `load(url)` it'll be loaded only once.

**Reply by ApayRus** [reply]
Thank you. Why 'potentially'? You give me hope to avoid double loading :)

**Reply by katspaugh** [reply]
Depends on the audio element params, you can set `preload="none"`, then it will start loading only on play.

**Reply by ApayRus** [reply]
Thank you. It works fine... 
@katspaugh I noticed only today your project 'videobook'. I had the similar idea, and realized it partly. I made parser for vtt subs, and extended them with additional elements, like chapters, so you could read a movie like a book, with a little player on the top. Also I added parallel translation. I read the movie 'Moana' in that way :) with my children. 
I was so happy when you merged my pull request after a year when I pushed it, because it was about subtitles, now I know )) 
I suppose we are like-minded people. 
![Screenshot from 2021-07-17 20-18-00](https://github.com/katspaugh/wavesurfer.js/assets/1222611/e729611a-29b0-4456-9950-39988982ecc6)

**Reply by katspaugh** [reply]
Nice app! Using technology for language learning is an amazing hobby, and it’s really cool that you’re also putting it to practice with your family.
Sorry it took so long to merge your PR btw, I hope we’ll see more of your contributions in future. Cheers!

**Reply by ApayRus** [reply]
My favorite movie (series) is Silicon Valley. I watched it for 5-6 times )) 
My dream project is to create movie reader, with: 

* navigation between series, and chapters inside series. 
* Highlight phrases/words, add bookmarks, comments... 

Each chapter (part of 1 episode) I want to provide with the title, cover image and maybe short description )) When I love something, I want to learn it deeply as possible )) So  it will be a real e-book, however connected to the movie, like navigable subs. 

I like the idea to split movie into chapters/scenes (inside the subs, vtt allows it, e.g. by comments). It makes it structured like a book or comics. And you see logical schema underneath fast moving images. Also it is interesting to see each chapter as separate video, with short length (e.g. 2 minutes), and it is easy to navigate inside it, because seekbar for this 2 minutes takes whole player width.
